src/dune_cache/config.ml

```diff
@@ -43,4 +43,5 @@ type t =
   | Enabled of
       { storage_mode : Dune_cache_storage.Mode.t
       ; reproducibility_check : Reproducibility_check.t
+      ; cache_user_rules : bool
```

Comment:
Whether the user rules are cached is not something that the cache itself should be aware of. It's a property of our rules and how we choose to create them.

Reply:
Good point, it the user rules should just state `can_go_in_shared_cache = true` in that case.